doc/workshops/plugin-development.rst

```diff
@@ -2,4 +2,1275 @@
+
+Propagate configuration changes
+-------------------------------
+
+Since we moved all the configuration information to ``conf.py``, now we have to import it from there into ``widgets.py``.
+
+`v0.0.4dev-v0.0.5dev widgets.py diff`_
+
+.. _v0.0.4dev-v0.0.5dev widgets.py diff: https://github.com/map0logo/spyder-pomodoro-timer/compare/v0.0.4-dev0...v0.0.5-dev0#diff-f156d182eb57c9fa20d1ca4b9677d13b158447c04f90724d936bb77a50df31bc
+
+.. code-block:: python
+
+   # Local imports
+   from spyder_pomodoro_timer.spyder.config import (
+       CONF_SECTION,
+       CONF_DEFAULTS,
+       CONF_VERSION,
+   )
+
+Once we have the form ready, we can access the configuration value from anywhere in our plugin using the ``get_conf`` method, in this case we use it to access the value of the configurable parameter ``pomodoro_limit`` from the configuration instead of the constant ``POMODORO_DEFAULT``.
+
+.. code-block:: python
+
+           self.pomodoro_limit = self.get_conf(
+               "pomodoro_limit"
+           )
+
+Now we can add a method that updates our configurable parameter ``pomodoro_limit``. The ``@on_conf_change`` decorator is the one in charge of capturing the signal that is generated when applying the change of a specific configuration parameter.
+
+.. code-block:: python
+
+       @on_conf_change(option="pomodoro_limit")
+       def set_pomodoro_limit(self, value):
+           self.pomodoro_limit = int(value) * 1000 * 60
+           self.countdown = self.pomodoro_limit
+           self.value = self.display_time()
+
+Registering preferences
+-----------------------
+
+Finally, it is necessary to activate the use of the preferences from ``plugin.py``, including the preferences as a requirement,
+
+`v0.0.4dev-v0.0.5dev plugin.py diff`_
+
+.. _v0.0.4dev-v0.0.5dev plugin.py diff: https://github.com/map0logo/spyder-pomodoro-timer/compare/v0.0.4-dev0...v0.0.5-dev0#diff-1e1feff1c9cd8690a64077ba61b59db060cae1212f9da8ce0522d3e027d09b0d
+
+.. code-block:: python
+
+   class SpyderPomodoroTimer(SpyderPluginV2):
+       ...
+       REQUIRES = [Plugins.Preferences, Plugins.StatusBar, Plugins.Toolbar]
+
+
+and registering them in a method with the decorator ``@on_plugin_available``.
+
+.. code-block:: python
+
+       @on_plugin_available(plugin=Plugins.Preferences)
+       def on_preferences_available(self):
+           preferences = self.get_plugin(Plugins.Preferences)
+           preferences.register_plugin_preferences(self)
+
+Now we can access the Preferences window either from the toolbar or from the "Tools > Preferences" menu and we will find a section *Spyder Pomodoro Timer* and inside it is the *Pomodoro timer limit* parameter, if we change the value and apply the change we will see how the corresponding label in the status bar changes.
+
+Now your plugin is in an initial version ready to publish...
```

Comment:
Maybe it is worth having a final screenshot with all the elements you added to spyder with this plugin pointed out.